### PR TITLE
Queue |  Prevent clearing all issue dispositions when editing an issue

### LINKED ISSUE #5402: Prevent clearing all issue dispositions when editing an issue
When an attorney edited an issue today, all other issues' dispositions were cleared. We should persist issue dispositions for all issues despite the user taking an action, like editing an issue, during check out.

## AC
- When a user edits one issue and then navigates back to the Select Dispositions screen, all previously set issue dispositions remain.

### PR DESCRIPTION
**Pull request by dannysellers**
Resolves #5402 

### Description
After creating/updating an issue, we overwrite the staged appeal's issues with the server's version (i.e. to get the correct server-assigned ID after creation). However, because we haven't updated dispositions yet, we'll end up overwriting dispositions set during the checkout flow.

### Acceptance Criteria 
- [ ] Dispositions set before creating/editing an issue are preserved

### Testing Plan
1. Go to Draft Decision checkout flow, set issue dispositions
2. Create or edit an issue, confirm return to Select Dispositions view
3. Confirm previously-set dispositions are still set

dispositions_preserved

### CODE REVIEW

File: client/app/queue/AddEditIssueView.jsx [hunk 1]
```diff
@@ -105,7 +105,7 @@ class AddEditIssueView extends React.Component {
           level_1: _.get(issue.codes, 0, null),
           level_2: _.get(issue.codes, 1, null),
           level_3: _.get(issue.codes, 2, null),
-          ..._.omit(issue, 'type', 'codes')
+          ..._.omit(issue, 'type', 'codes', 'disposition')
```
Comment (mdbenjam) [inline client/app/queue/AddEditIssueView.jsx]: Ah interesting, so we could have erroneously set dispositions on the edit issue screen.

File: client/app/queue/components/SelectIssueDispositionDropdown.jsx [hunk 2]
```diff
@@ -55,7 +55,7 @@ class SelectIssueDispositionDropdown extends React.PureComponent {
           marginTop: '1rem'
         })}
         onChange={(readjudication) => this.props.updateIssue({ readjudication })}
-        value={issue.readjudication}
+        value={issue.readjudication || false}
```
Comment (mdbenjam) [inline client/app/queue/components/SelectIssueDispositionDropdown.jsx]: curious what is the motivation for this change? Is readjudication not always defined?

File: client/app/queue/AddEditIssueView.jsx [hunk 3]
```diff
@@ -122,11 +122,26 @@ class AddEditIssueView extends React.Component {
...
-      this.props.saveEditedAppealIssue(this.props.vacolsId, JSON.parse(resp.text))
-    );
+    requestPromise.then((resp) => this.updateIssuesFromServer(JSON.parse(resp.text)));
   };
 
+  updateIssuesFromServer = (response) => {
+    const { appeal: { attributes: appeal } } = this.props;
+    const serverIssues = response.issues;
+
+    const issues = _.map(serverIssues, (issue) => {
```
Comment (mdbenjam) [inline client/app/queue/AddEditIssueView.jsx]: Do we need to go through this? Can't we just drop all the dispositions from the response? Shouldn't all the dispositions be nil from the server?

File: client/app/queue/AddEditIssueView.jsx (hunk 1 above)
Comment (dannysellers) [reply client/app/queue/AddEditIssueView.jsx]: Oh, no, we don't need this after all. [`IssuesController.issue_params`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/controllers/issues_controller.rb#L59) doesn't accept the `disposition` parameter. It'd be better to change this to `_.pick`

File: client/app/queue/components/SelectIssueDispositionDropdown.jsx (hunk 2 above)
Comment (dannysellers) [reply client/app/queue/components/SelectIssueDispositionDropdown.jsx]: Readjudication isn't a property included in [`IssueSerializer`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/serializers/work_queue/issue_serializer.rb), so it's initially `null`, and we get the 'uncontrolled to controlled component' warning when the user sets the state

File: client/app/queue/AddEditIssueView.jsx (hunk 3 above)
Comment (dannysellers) [reply client/app/queue/AddEditIssueView.jsx]: We don't support judge checkout yet, but this is for the user who's reviewing a case with issues that have dispositions already set: if they make changes to dispositions during checkout, those changes would be overwritten with server data after the user creates/updates an issue.

You're right that we never need to use dispositions from the response, but we want to use everything else that comes in

File: client/app/queue/AddEditIssueView.jsx [hunk 4]
```diff
@@ -122,11 +122,26 @@ class AddEditIssueView extends React.Component {
...
+      // preserve locally-updated dispositions
+      const disposition = _.get(
+        _.find(appeal.issues, (iss) => iss.vacols_sequence_id === issue.vacols_sequence_id),
+        'disposition'
+      );
+
+      return _.defaults({ disposition }, issue);
```
Comment (mdbenjam) [inline client/app/queue/AddEditIssueView.jsx]: nit: Did not know what _.defaults did. Had to look it up. Seems like it just does the opposite of normal ES6 syntax. Would this also work? [...issue, { disposition }]? I feel that's clearer, but might just be personal preference.
